components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/VariantListController.java

```diff
@@ -0,0 +1,407 @@
+            if (field != null) {
+                String value = "";
+                switch (property) {
+                    case INTERPRETATION_KEY:
+                        value = parseInterpretation(field.getValue());
+                        break;
+                    case RESOLUTION_KEY:
+                        value = parseResolution(field.getValue());
+                        break;
+                    case SEGREGATION_KEY:
+                        value = parseSegregation(field.getValue());
+                        break;
+                    case SANGER_KEY:
+                        value = parseSegregation(field.getValue());
+                        break;
+                    case EVIDENCE_KEY:
+                        value = parseEvidence(field.getValue());
+                        break;
+                    case EFFECT_KEY:
+                        value = parseEffect(field.getValue());
+                        break;
+                    case INHERITANCE_KEY:
+                        value = parseInheritance(field.getValue());
+                        break;
+                    default:
+                        value = field.getValue();
+                        break;
+                }
+                singleVariant.put(property, value);
+            }
+        }
+        return singleVariant;
+    }
+
+    @Override
+    public PatientData<Map<String, String>> load(Patient patient)
+    {
+        try {
+            XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
+            List<BaseObject> variantXWikiObjects = doc.getXObjects(VARIANT_CLASS_REFERENCE);
+            if (variantXWikiObjects == null || variantXWikiObjects.isEmpty()) {
+                return null;
+            }
+
+            List<Map<String, String>> allVariants = new LinkedList<Map<String, String>>();
+            for (BaseObject variantObject : variantXWikiObjects) {
+                if (variantObject != null) {
+                    Map<String, String> singleVariant = parseFields(variantObject);
+                    allVariants.add(singleVariant);
+                }
+            }
+            return new IndexedPatientData<Map<String, String>>(getName(), allVariants);
+        } catch (Exception e) {
+            this.logger.error("Could not find requested document or some unforeseen "
+                + "error has occurred during controller loading ", e.getMessage());
+        }
+        return null;
+    }
+
+    private void removeKeys(Map<String, String> item, List<String> keys, List<String> enablingProperties,
+        Collection<String> selectedFieldNames)
+    {
+        int count = 0;
+        for (String property : keys) {
+            if (StringUtils.isBlank(item.get(property))
+                || (selectedFieldNames != null
+                && !selectedFieldNames.contains(enablingProperties.get(count)))) {
+                item.remove(property);
+            }
+            count++;
```

Comment:
Rather than implementing `enablingProperties` as a list, perhaps it would be better to implement it as a map of property_name --> enabling_field_name. That way this code is not dependent on the indices of the `keys` list and the `enablingProperties` list matching up